### PR TITLE
[ET-2014] Fix - Removed strict type casting from signature to avoid fatals with external filtered values.

### PR DESCRIPTION
**Pull request by rafsuntaskin**
### 🎫 Ticket

[ET-2014] 

### 🗒️ Description

* We are having some fatal reports from external plugins/extensions with the strict type casting on a filtering method. This PR removes the strict type casting from the signature to avoid that.
* You can use this `mu-plugin` to test the fix: https://gist.github.com/rafsuntaskin/ed47584d25ecad0365175e68e8f1df66

### 🎥 Artifacts 
🎥 screencast(s): https://www.loom.com/share/f146d5a6084041f5b56e0910bdafbd41

### ✔️ Checklist
- [ ] I've included a changelog entry in the readme.txt file. 
- [ ] My code is tested. 
- [ ] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

[ET-2014]: https://stellarwp.atlassian.net/browse/ET-2014?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tickets/Flexible_Tickets/Series_Passes/Base.php [hunk 1]
```diff
@@ -603,23 +603,24 @@ public function hide_header_image_option_from_ticket_settings( string $html = nu
...
 	 * @param WP_Post $series       The series post object.
-	 * @param bool    $display_time Whether to display the time.
-	 * @param string  $date_format  The date format.
+	 * @param mixed   $display_time Whether to display the time.
+	 * @param mixed   $date_format  The date format.
 	 *
 	 * @return string The updated start date.
 	 */
-	public function filter_start_date_for_series( string $start_date, WP_Post $series, bool $display_time, string $date_format ): string {
+	public function filter_start_date_for_series( string $start_date, WP_Post $series, $display_time, $date_format ): string {
```
Comment (lucatume) [inline src/Tickets/Flexible_Tickets/Series_Passes/Base.php:615]: I would remove the assumption the `start_date` parameter is a `string` as well.

File: src/Tickets/Flexible_Tickets/Series_Passes/Base.php [hunk 2]
```diff
@@ -603,23 +603,24 @@ public function hide_header_image_option_from_ticket_settings( string $html = nu
 	 * Filters the start date for a series to use the start date of the first event in the series.
 	 *
 	 * @since 5.8.0
+	 * @since TBD Changed the strict type of the $display_time and $date_format parameter to mixed.
 	 *
 	 * @param string  $start_date   The start date.
 	 * @param WP_Post $series       The series post object.
-	 * @param bool    $display_time Whether to display the time.
-	 * @param string  $date_format  The date format.
+	 * @param mixed   $display_time Whether to display the time.
```
Comment (lucatume) [inline src/Tickets/Flexible_Tickets/Series_Passes/Base.php:610]: I would not change the types in the doc-block: they represent what is _ideally_ expected.

File: src/Tickets/Flexible_Tickets/Series_Passes/Base.php (hunk 1 above)
Comment (lucatume) [reply src/Tickets/Flexible_Tickets/Series_Passes/Base.php:615]: Default all parameters to `null`

File: src/Tickets/Flexible_Tickets/Series_Passes/Base.php [hunk 3]
```diff
@@ -603,23 +603,24 @@ public function hide_header_image_option_from_ticket_settings( string $html = nu
...
 		if ( get_post_type( $series ) !== Series_Post_Type::POSTTYPE ) {
 			return $start_date;
 		}
-
+		
 		$first_event = tribe_events()->where( 'series', $series->ID )
-		                             ->order_by( 'event_date', 'ASC' )
-		                             ->per_page( - 1 )
-		                             ->first();
+									->order_by( 'event_date', 'ASC' )
```
Comment (lucatume) [inline src/Tickets/Flexible_Tickets/Series_Passes/Base.php:621]: ???

File: src/Tickets/Flexible_Tickets/Series_Passes/Base.php [hunk 4]
```diff
@@ -632,23 +633,24 @@ public function filter_start_date_for_series( string $start_date, WP_Post $serie
 	 * Filters the end date for a series to use the start date of the last event in the series.
 	 *
 	 * @since 5.8.0
+	 * @since TBD Changed the strict type of the $display_time and $date_format parameter to mixed.
 	 *
 	 * @param string  $end_date     The end date.
 	 * @param WP_Post $series       The series post object.
-	 * @param bool    $display_time Whether to display the time.
-	 * @param string  $date_format  The date format.
+	 * @param mixed   $display_time Whether to display the time.
```
Comment (lucatume) [inline src/Tickets/Flexible_Tickets/Series_Passes/Base.php:640]: Do not change the types in the doc-block.

File: src/Tickets/Flexible_Tickets/Series_Passes/Base.php [hunk 5]
```diff
@@ -632,23 +633,24 @@ public function filter_start_date_for_series( string $start_date, WP_Post $serie
...
+	 * @param mixed   $date_format  The date format.
 	 *
 	 * @return string The updated end date.
 	 */
-	public function filter_end_date_for_series( string $end_date, WP_Post $series, bool $display_time, string $date_format ): string {
+	public function filter_end_date_for_series( string $end_date, WP_Post $series, $display_time, $date_format ): string {
```
Comment (lucatume) [inline src/Tickets/Flexible_Tickets/Series_Passes/Base.php:645]: Remove the `string` type-hint from the `$end_date` parameter.

Comment (lucatume) [reply src/Tickets/Flexible_Tickets/Series_Passes/Base.php:645]: In the methods, make sure the values are of the expected type, else bail out.

File: src/Tickets/Flexible_Tickets/Series_Passes/Base.php (hunk 1 above)
Comment (lucatume) [reply src/Tickets/Flexible_Tickets/Series_Passes/Base.php:615]: In the method, make sure the parameters are of the expected type, else bailout.

**Review by lucatume — CHANGES_REQUESTED** [review]

File: src/Tickets/Flexible_Tickets/Series_Passes/Base.php (hunk 3 above)
Comment (rafsuntaskin) [reply src/Tickets/Flexible_Tickets/Series_Passes/Base.php:621]: phpcbf -_-

**Review by lucatume — APPROVED** [review]

**Review by redscar — APPROVED** [review]

**Pull request merged**